third_party/openvino/ngraph_c_api/src/ngraph_c_api.cpp

```diff
@@ -0,0 +1,472 @@
+                        ngraph_node_t** node) {
+  BUILD_BINARY(v1::Subtract, a, b, node);
+}
+
+IEStatusCode ngraph_leaky_relu(const ngraph_node_t* a,
+                               const ngraph_node_t* b,
+                               ngraph_node_t** node) {
+  BUILD_BINARY(v0::PRelu, a, b, node);
+}
+
+IEStatusCode ngraph_mat_mul(const ngraph_node_t* a,
+                            const ngraph_node_t* b,
+                            ngraph_node_t** node) {
+  auto matmul = std::make_shared<ngraph::op::v0::MatMul>(a->object, b->object,
+                                                         false, false);
+  CREATE_NGRAPH_NODE(matmul, node);
+}
+
+IEStatusCode ngraph_transpose(const ngraph_node_t* a,
+                              const ngraph_node_t* b,
+                              ngraph_node_t** node) {
+  BUILD_BINARY(v1::Transpose, a, b, node);
+}
+
+IEStatusCode ngraph_reshape(const ngraph_node_t* a,
+                            const ngraph_node_t* b,
+                            ngraph_node_t** node) {
+  auto reshape =
+      std::make_shared<ngraph::op::v1::Reshape>(a->object, b->object, true);
+  CREATE_NGRAPH_NODE(reshape, node);
+}
+
+IEStatusCode ngraph_max(const ngraph_node_t* a,
+                        const ngraph_node_t* b,
+                        ngraph_node_t** node) {
+  BUILD_BINARY(v1::Maximum, a, b, node);
+}
+
+IEStatusCode ngraph_min(const ngraph_node_t* a,
+                        const ngraph_node_t* b,
+                        ngraph_node_t** node) {
+  BUILD_BINARY(v1::Minimum, a, b, node);
+}
+
+IEStatusCode ngraph_power(const ngraph_node_t* a,
+                          const ngraph_node_t* b,
+                          ngraph_node_t** node) {
+  BUILD_BINARY(v1::Power, a, b, node);
+}
+
+IEStatusCode ngraph_divide(const ngraph_node_t* a,
+                           const ngraph_node_t* b,
+                           ngraph_node_t** node) {
+  BUILD_BINARY(v1::Divide, a, b, node);
+}
+
+IEStatusCode ngraph_relu(const ngraph_node_t* a, ngraph_node_t** node) {
+  BUILD_UNARY(v0::Relu, a, node);
+}
+
+IEStatusCode ngraph_softmax(const ngraph_node_t* a, ngraph_node_t** node) {
+  auto softmax = std::make_shared<ngraph::op::v1::Softmax>(a->object, 1);
+  CREATE_NGRAPH_NODE(softmax, node);
+}
+
+IEStatusCode ngraph_sigmoid(const ngraph_node_t* a, ngraph_node_t** node) {
+  BUILD_UNARY(v0::Sigmoid, a, node);
+}
+
+IEStatusCode ngraph_tanh(const ngraph_node_t* a, ngraph_node_t** node) {
+  BUILD_UNARY(v0::Tanh, a, node);
+}
+
+IEStatusCode ngraph_concat(ngraph_node_t** inputs,
+                           uint32_t input_count,
+                           uint32_t axis,
+                           ngraph_node_t** node) {
+  ngraph::OutputVector inputs_vector;
+  inputs_vector.reserve(input_count);
+  for (uint32_t i = 0; i < input_count; ++i) {
+    inputs_vector.push_back(inputs[i]->object);
+  }
+  auto concat = std::make_shared<ngraph::op::v0::Concat>(inputs_vector, axis);
```

Comment:
My general question is where to try and catch the ngraph exception. I found it is useful to know the ngraph error message when debugging. Previously we have `BEGINE_TRY` and `END_CATCH`.

Reply:
Thanks, done.